src/utils/calendar.ts

```diff
@@ -0,0 +1,34 @@
+import { PersonalizedVenueEvent } from "types/venues";
+import { VenueEvent } from "types/venues";
+import { eventEndTime, eventStartTime } from "utils/event";
+import { WithVenueId } from "utils/id";
+import { getFullVenueInsideUrl } from "utils/url";
+
+import ical from "ical-generator";
+
+export const createCalendar = (
+  allEvents: PersonalizedVenueEvent[] | WithVenueId<VenueEvent>[],
+  calendarname: string
+) => {
+  const cal = ical({ name: calendarname });
+
+  if (!cal) return;
+
+  allEvents.map((event: PersonalizedVenueEvent | WithVenueId<VenueEvent>) =>
+    cal.createEvent({
+      id: event.venueId,
```

Comment:
I think we'll need to use something different for `id` here, since `event.venueId` isn't going to be unique.

I don't believe we have a unique `id` representing our events currently. Not sure what to use here, as i'm not sure what the calendar 'does with it'.

Further Reading:

- https://sebbo2002.github.io/ical-generator/develop/reference/classes/icalevent.html#id

---

@mike-lvov Any thoughts?

Reply:
I will remove it for now since you mention it's not a unique number

Reply:
Sounds good. Does the `ical` lib need us to set it? Or will it just generate its own internally if required? Given it depends on a `uuid` lib, i'm guessing it will probably generate it on it's own if needed.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1423/commits/ec68e834fab42e9549a285d7a454e4c29e1f24e4